### ISSUE TITLE
Documentation for embedding VM

### ISSUE THREAD

**Issue by elgunhuseynli** [dart-lang/sdk#37527]
Hi.

Our team is interested in embedding Dart VM in our game engine for scripting. Unfortunately we couldn't found any sample or documentation about that.

Currently our engine has scripting support with Lua. But we want to replace Lua with Dart. Any idea about how to do it?

Thanks.

**Comment by chancila** [comment]
I figured a good amount of this on my own reading the main dart embedder source code:

https://gist.github.com/chancila/f0b4e1b37f44fa780427711e286abdad

see https://github.com/chancila/sdk for the build system change you'll need

it would be really nice if dart came out of the box with something like this

**Comment by kwalrath** [comment]
Would [dart2aot](https://dart.dev/tools/dart2aot) or its impending replacement, [dart2native](https://sz-www-1.firebaseapp.com/tools/dart2native), do?

**Comment by chancila** [comment]
I don't think so, both of those generate an executable, I was looking for functionality similar to v8's embedding story which is better documented and supported in the build, which enables use of v8 as a subsystem in a larger program. v8's build basically spits out a library you can link which is the only thing you need in order to consume the embedder api.

 I think flutter pretty much consumes dart in this fashion if I'm not mistaken. It's not impossible to do at the moment it's just undocumented and not streamlined, and becomes even harder if you don't use gn.

**Comment by whaqzhzd** [comment]
How to compile dart VM as static library
